src/zks_provider/mod.rs

```diff
@@ -335,11 +377,70 @@ impl<M: Middleware + ZKSProvider, S: Signer> ZKSProvider for SignerMiddleware<M,
     ) -> Result<DebugTrace, ProviderError> {
         ZKSProvider::debug_trace_transaction(self.inner(), hash, options).await
     }
+
+    async fn send_eip712<T, D>(
+        &self,
+        wallet: &Wallet<D>,
+        contract_address: Address,
+        function_signature: &str,
+        function_parameters: Option<T>,
+        overrides: Option<Overrides>,
+    ) -> Result<(Vec<Token>, H256), ProviderError>
+    where
+        T: Tokenizable + Debug + Clone + Send,
+        D: PrehashSigner<(RecoverableSignature, RecoveryId)> + Send + Sync,
+    {
+        self.inner()
+            .send_eip712(
+                wallet,
+                contract_address,
+                function_signature,
+                function_parameters,
+                overrides,
+            )
+            .await
+    }
+
+    async fn send<T, D>(
+        &self,
+        wallet: &Wallet<D>,
+        contract_address: Address,
+        function_signature: &str,
+        function_parameters: Option<T>,
+        _overrides: Option<Overrides>,
+    ) -> Result<(Vec<Token>, H256), ProviderError>
+    where
+        T: Tokenizable + Debug + Clone + Send,
+        D: PrehashSigner<(RecoverableSignature, RecoveryId)> + Send + Sync,
+    {
+        let tx = build_send_tx(
+            self,
+            wallet.address(),
+            contract_address,
+            function_signature,
+            function_parameters,
+            _overrides,
+        )
+        .await?;
+        let pending_transaction = self.send_transaction(tx, None).await.map_err(|e| {
+            ProviderError::CustomError(format!("Error sending transaction: {:?}", e))
+        })?;
+
+        // TODO: Should we wait here for the transaction to be confirmed on-chain?
```

Comment:
I think we should. The python implementation has a polling mechanism, right?

Reply:
Yeah, we definitely need it. That will avoid having to manually add a sleep before the finalize. I think I will open a new PR with this functionality since it is necessary for all other wallet operations.